client/state/selectors/get-current-plan-term.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+
+import { getPlan } from 'lib/plans';
+import { getSitePlan } from 'state/sites/selectors';
+
+/**
+ * Returns term of the active plan for given siteId, e.g. TERM_MONTHLY
+ *
+ * @param {Object} state Current state
+ * @param {Number} siteId Site ID
+ * @return {string} Human-readable interval type
+ */
+export default function getTermFromCurrentPlan( state, siteId ) {
+	const currentPlanProduct = getSitePlan( state, siteId );
+	if ( ! currentPlanProduct ) {
+		return null;
+	}
+
+	const currentPlanObject = getPlan( currentPlanProduct.product_slug );
```

Comment:
Seems like this selector could use some caching (`createSelector()`).

Reply:
How costly are `getSitePlan` and `getPlan` expected to be?

Reply:
Not costly at all :-)

Reply:
`getPlan` should be pretty cheap in the usual use case. 

`getSitePlan` is generating some objects on every call, and it lacks caching. Perhaps it needs to have caching setup internally, which is a different task anyway.

In any case it's worth doing an audit to see if we could improve the performance by introducing some sort of caching or memoization.